syntax = "proto3";

package svg_proto;

message Point {
    double x = 1;
    double y = 2;
}

message Rgb {
    uint32 red = 1;
    uint32 green = 2;
    uint32 blue = 3;
}

message Rgba {
    uint32 red = 1;
    uint32 green = 2;
    uint32 blue = 3;
    double opacity = 4;
}

message Color {
    oneof color {
        string name = 1;
        Rgb rgb = 2;
        Rgba rgba = 3;
    }
}